tests/manage/pv_services/test_resource_deletion_during_pvc_pod_deletion_and_io.py

```diff
@@ -0,0 +1,335 @@
+
+        # Get number of pods of type 'resource_to_delete'
+        num_of_resource_to_delete = len(pod_functions[resource_to_delete]())
+
+        # Fetch the number of Pods and PVCs
+        initial_num_of_pods = len(get_all_pods(namespace=namespace))
+        initial_num_of_pvc = len(
+            get_all_pvcs(namespace=namespace)['items']
+        )
+
+        # Fetch PV names to verify after deletion
+        pv_objs = []
+        for pvc_obj in pvcs_to_delete:
+            pvc_obj.reload()
+            pv_objs.append(pvc_obj.backed_pv_obj)
+
+        # Fetch volume details from pods for the purpose of verification
+        node_pv_dict = {}
+        for pod_obj in pods_to_delete:
+            pod_info = pod_obj.get()
+            node = pod_info['spec']['nodeName']
+            pvc = pod_info['spec']['volumes'][0]['persistentVolumeClaim']['claimName']
+            for pvc_obj in pvc_objs:
+                if pvc_obj.name == pvc:
+                    pvc_obj.reload()
+                    pv = pvc_obj.backed_pv
+                    break
+            if node in node_pv_dict:
+                node_pv_dict[node].append(pv)
+            else:
+                node_pv_dict[node] = [pv]
+
+        # Fetch image uuid associated with PVCs to be deleted
+        pvc_uuid_map = {}
+        for pvc_obj in pvcs_to_delete:
+            pvc_uuid_map[pvc_obj.name] = pvc_obj.image_uuid
+        log.info("Fetched image uuid associated with each PVC")
+
+        # Do setup on pods for running IO
+        log.info("Setting up pods for running IO.")
+        for pod_obj in pod_objs:
+            executor.submit(pod_obj.workload_setup, storage_type='fs')
+
+        # Wait for setup on pods to complete
+        for pod_obj in pod_objs:
+            for sample in TimeoutSampler(
+                100, 2, getattr, pod_obj, 'wl_setup_done'
+            ):
+                if sample:
+                    log.info(
+                        f"Setup for running IO is completed on pod "
+                        f"{pod_obj.name}."
+                    )
+                    break
+        log.info("Setup for running IO is completed on all pods.")
+
+        # Start IO on pods having PVCs to delete to load data
+        log.info("Starting IO on pods having PVCs to delete.")
+        for pod_obj in pods_for_pvc:
+            pod_obj.run_io(storage_type='fs', size=f'{self.pvc_size - 1}G')
+        log.info("IO started on pods having PVCs to delete.")
+
+        log.info("Fetching IO results from the pods having PVCs to delete.")
+        for pod_obj in pods_for_pvc:
+            get_fio_rw_iops(pod_obj)
+        log.info("Verified IO result on pods having PVCs to delete.")
+
+        # Delete pods having PVCs to delete.
+        assert self.delete_pods(pods_for_pvc), (
+            "Couldn't delete pods which are having PVCs to delete."
+        )
+        for pod_obj in pods_for_pvc:
+            assert pod_obj.ocp.wait_for_delete(pod_obj.name), (
```

Comment:
ocp.wait_for_delete() is not returning True or False so this will always fail

Reply:
It will return bool only on success.
Removed assert